libraries/redcore/api/hal/hal.php

```diff
@@ -628,6 +628,26 @@ public function apiCreate()
 		$functionName = RApiHalHelper::attributeToString($this->operationConfiguration, 'functionName', 'save');
 		$data = $this->triggerFunction('processPostData', $this->options->get('data', array()), $this->operationConfiguration);
 
+		// Remove primary keys for "model" and "table" data mode
+		$dataMode = RApiHalHelper::attributeToString($this->operationConfiguration, 'dataMode');
+
+		if (($dataMode == 'model') || ($dataMode == 'table'))
```

Comment:
I dont think it matters what the dataMode is... if it is create operation it should remove the primary key field